src/widget/wcolorpicker.cpp

```diff
@@ -30,96 +54,155 @@ WColorPicker::WColorPicker(ColorOption colorOption, QWidget* parent)
-            );
-        } else {
-            pColorButton->setProperty("noColor", true);
-        }
+    int numColors = m_palette.size();
+    if (m_colorOption == ColorOption::AllowNoColor) {
+        numColors++;
+    }
 
-        pColorButton->setToolTip(pColor->m_sDisplayName);
-        pColorButton->setCheckable(true);
-        m_pColorButtons.insert(pColor, pColorButton);
+    int numColumns = idealColumnCount(numColors);
+    if (m_colorOption == ColorOption::AllowNoColor) {
+        addColorButton(std::nullopt, pLayout, row, column);
+        column++;
+    }
 
-        pLayout->addWidget(pColorButton, row, column);
+    for (const auto& color : m_palette) {
+        addColorButton(color, pLayout, row, column);
         column++;
-        if (column == kNumColumns) {
+        if (column == numColumns) {
             column = 0;
             row++;
         }
-
-        connect(this,
-                &WColorPicker::colorPicked,
-                this,
-                &WColorPicker::slotColorPicked);
-        connect(pColorButton,
-                &QPushButton::clicked,
-                this,
-                [pColor, this]() {
-                    emit colorPicked(pColor);
-                });
     }
-    setLayout(pLayout);
 }
 
-void WColorPicker::setSelectedColor(PredefinedColorPointer pColor) {
-    if (m_pSelectedColor) {
-        auto it = m_pColorButtons.constFind(m_pSelectedColor);
-        if (it != m_pColorButtons.constEnd()) {
-            it.value()->setChecked(false);
-            // This is needed to re-apply skin styles (e.g. to show/hide a checkmark icon)
-            it.value()->style()->unpolish(it.value());
-            it.value()->style()->polish(it.value());
-        }
+void WColorPicker::addColorButton(mixxx::RgbColor::optional_t color, QGridLayout* pLayout, int row, int column) {
+    parented_ptr<QPushButton> pColorButton = make_parented<QPushButton>("", this);
+    if (m_pStyle) {
+        pColorButton->setStyle(m_pStyle);
     }
 
-    if (pColor) {
-        auto it = m_pColorButtons.constFind(pColor);
-        if (it != m_pColorButtons.constEnd()) {
-            it.value()->setChecked(true);
-            // This is needed to re-apply skin styles (e.g. to show/hide a checkmark icon)
-            it.value()->style()->unpolish(it.value());
-            it.value()->style()->polish(it.value());
+    if (color) {
+        // Set the background color of the button. This can't be overridden in skin stylesheets.
+        pColorButton->setStyleSheet(
+                QString("QPushButton { background-color: %1; }").arg(mixxx::RgbColor::toQString(color)));
+    } else {
+        pColorButton->setProperty("noColor", true);
+    }
+    pColorButton->setToolTip(mixxx::RgbColor::toQString(color, tr("No Color")));
+
+    pColorButton->setCheckable(true);
+    m_colorButtons.append(pColorButton);
+
+    pLayout->addWidget(pColorButton, row, column);
+
+    connect(this,
```

Comment:
something I've noticed recently: I don't really like this excessive linewrapping style that I've seen cropping up in our code.  I'd prefer:

```
connect(ob, signal,
        ob, slot);
```

if it doesn't fit on one line.  But if it fits on one line, I'd prefer one line

Reply:
Not possible unfortunately. `clang-format` only allows either all on one line or each param on a separate line (unless we specify a hard line length limit of 80 chars AFAIK, but that idea was rejected).

Reply:
gross, I didn't realize clang was doing that.  Why only the connect function?

Reply:
It's the case for every function AFAIK.

Reply:
that can't possibly be true, addWidget on the line above isn't separating newlines after every comma.  Anyway not worth figuring out here, but something is fishy

Reply:
As I said, it is possible to have *all* arguments on one line, but type-safe `connect()` usually has long arguments so splitting is preferable.

Reply:
What is not possible is to put 4 arguments on 2 lines.

Reply:
Please don't open up discussions about code style again. Let's just accept the two choices that clang-format offers us without a line limit:
- All arguments on a single line
- All arguments on separate lines

In the end this is all just code, not a poem ;)